### PR TITLE
Migrated it to firefox

### PR DESCRIPTION
**Pull request by vaibhavdevnani7**
Migrated the extension to make it support firefox.

### CODE REVIEW

**Comment by Rutuj-Runwal** [comment]
@vaibhavdevnani7 You have removed the declarativeNstRequest code andchanged it to the webrequest api.

Though this works on Firefox but the extension is no longer in Manifest V3 which is the new standard for browser extensions. Moreover, chrome and edge have stopped accepting new extensions based on Manifest V2 in their stores.

**Comment by vaibhavdevnani7** [comment]
Yeah, that was the idea, keep firefox under a separate branch, and then merge it with the main when firefox adopts MV3.

**Comment by Rutuj-Runwal** [comment]
> Yeah, that was the idea, keep firefox under a separate branch, and then merge it with the main when firefox adopts MV3.

Great! 
You can wait till Matt reviews it.
Till then you can contribute here: https://github.com/Rutuj-Runwal/RR-Adblocker/issues/17 
We need a firefox migration too.

**Comment by immattdavison** [comment]
Hey @vaibhavdevnani7 the idea was not to migrate to Firefox but to build a new version which does work with Firefox, this may mean it is necessary to restructure the file tree and put some stuff in different folders. I unfortunately cannot merge this as it will no longer work for Google Chrome if I do so.

File: blockpage.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ async function retryOnTabUpdate(tabId, info, tab) {
   }
 }
 
-chrome.runtime.onConnect.addListener((port) => {
+browser.runtime.onConnect.addListener((port) => {
```
Comment (immattdavison) [inline blockpage.js:39]: This all relies on chromes internal API for browser extensions and thus becomes disfunctional with these changes

File: blockpage.js [hunk 2]
```diff
@@ -49,43 +49,33 @@ function fetchProtectionRules(url,status){
   fetch(url)
   .then((res) => res.json())
   .then((domains) => {
+    for (var i=0; i<domains.length; i++){
+      domains[i]="*://www."+domains[i]+"/*"
```
Comment (immattdavison) [inline blockpage.js:53]: It should not rely on www, www is not as widely used as it was in the past as it is no longer necessary

**Review by immattdavison — CHANGES_REQUESTED** [review]

**Comment by vaibhavdevnani7** [comment]
This will work on any browser now, you can try on chrome

**Comment by Rutuj-Runwal** [comment]
> This will work on any browser now, you can try on chrome
> 

It will work but Manifest V2 is going to be deprecated and Google has stopped accepting extensions built on it in the chrome web store.

**Comment by Rutuj-Runwal** [comment]
> This is what I mean though, we can't move backwards, MV3 needs to be there for chrome so a separate dir may be better in this instance?

True, we cant move backwards
We can create separate dir or we can just create a separate branch and merge it in that branch and maintain the MV2 version there.

**Comment by immattdavison** [comment]
Personally I think we're better to either have a whole other repo, or a separate dir for each version, branches could get messy? 🤷‍♂️

**Comment by Rutuj-Runwal** [comment]
> Personally I think we're better to either have a whole other repo, or a separate dir for each version, branches could get messy? 🤷‍♂️

A whole.new repo wont make sense as MV2 is short-lived.Creating a dir is better choice then

@vaibhavdevnani7 so keep this current version as is. Make a new dir "NoMoreDomains-MV2" or something like that inside the repo and make your changes there. Add everything to the new manifest.json file,background scripts,UI etc.

**Comment by immattdavison** [comment]
Would it potentially be a good idea to put MV3 in its own dir as well then so it doesn't look as though MV2 is part of MV3?

**Comment by Rutuj-Runwal** [comment]
> Would it potentially be a good idea to put MV3 in its own dir as well then so it doesn't look as though MV2 is part of MV3?

Sure that will be nice we can keep all un necessary markdown files and contribution guidelines readme etc outside and create separate folders for both versions

**Pull request closed without merging**